src/test/java/io/strimzi/ValidatingWebhookTest.java

```diff
@@ -146,14 +147,117 @@ public void testEvictionOnPodWithExistingAnnotations() {
     }
 
     @Test
-    public void testWrongName() {
-        String podName = "my-cluster-connect-1";
+    public void testWrongLabel() {
+        String podName = "my-cluster-kafka-1";
+        Map label = Collections.singletonMap("mylabel", "foo");
         when(podResource.get()).thenReturn(mockedPod(podName, true, false));
```

Comment:
Pod name is now irrelevant and can be hard coded. Instead, the set of labels is what you want to test against the various webhook configurations.

Whatever label comes in a `reviewRequest`, should also be in the `mockedPod`, so it could help to factor out the map of labels and pass it as parameter to both methods.